src/code/code_800EC960.c

```diff
@@ -251,44 +251,542 @@ u8 sSeqFlags[0x6E] = {
-    { 0xC003, 0xC000, { 0, 2, 0, 0, 3, 0, 1, 2, 4, 1, 3, 0, 1, 4, 1, 1, 5, 16, 0xFF } },
-    { 0xC003, 0xC000, { 0, 2, 0, 0, 3, 0, 1, 2, 4, 1, 3, 0, 1, 4, 1, 1, 5, 16, 0xFF } },
+// clang-format off
+NatureAmbienceDataIO sNatureAmbienceDataIO[20] = {
+    // natureAmbienceId = NATURE_ID_GENERAL_NIGHT
+    {
+        0xC0FF, // PlayerIO Data
+        0xC0FE, // Channel Mask
+        {
+            // Channel 0
+            NATURE_CHANNEL_STREAM_0, CHANNEL_IO_PORT_2, NATURE_STREAM_RUSHING_WATER,
+            NATURE_CHANNEL_STREAM_0, CHANNEL_IO_PORT_3, 0,
+
+            // Channel 1
+            NATURE_CHANNEL_ANIMALS_0, CHANNEL_IO_PORT_2, NATURE_ANIMALS_CROWS_CAWS,
+            NATURE_CHANNEL_ANIMALS_0, CHANNEL_IO_PORT_3, 64, // bend pitch
+            NATURE_CHANNEL_ANIMALS_0, CHANNEL_IO_PORT_4, 0,  // number of layers
+            NATURE_CHANNEL_ANIMALS_0, CHANNEL_IO_PORT_5, 32, // reverb?
+
+            // Channel 2
+            NATURE_CHANNEL_ANIMALS_1, CHANNEL_IO_PORT_2, NATURE_ANIMALS_CRICKETS,
+            NATURE_CHANNEL_ANIMALS_1, CHANNEL_IO_PORT_3, 0,  // bend pitch
+            NATURE_CHANNEL_ANIMALS_1, CHANNEL_IO_PORT_4, 1,  // number of layers
+            NATURE_CHANNEL_ANIMALS_1, CHANNEL_IO_PORT_5, 16,
+
+            // Channel 3
+            NATURE_CHANNEL_ANIMALS_2, CHANNEL_IO_PORT_2, NATURE_ANIMALS_SMALL_BIRD_CHIRPS,
+            NATURE_CHANNEL_ANIMALS_2, CHANNEL_IO_PORT_3, 112, // bend pitch
+            NATURE_CHANNEL_ANIMALS_2, CHANNEL_IO_PORT_4, 1,   // number of layers
+            NATURE_CHANNEL_ANIMALS_2, CHANNEL_IO_PORT_5, 48,
+
+            // Channel 4
+            NATURE_CHANNEL_ANIMALS_3, CHANNEL_IO_PORT_2, NATURE_ANIMALS_HAWK_SCREECH,
+            NATURE_CHANNEL_ANIMALS_3, CHANNEL_IO_PORT_3, 127, // bend pitch
+            NATURE_CHANNEL_ANIMALS_3, CHANNEL_IO_PORT_4, 0,   // number of layers
+            NATURE_CHANNEL_ANIMALS_3, CHANNEL_IO_PORT_5, 16,
+
+            // Channel 5
+            NATURE_CHANNEL_ANIMALS_4, CHANNEL_IO_PORT_2, NATURE_ANIMALS_BIRD_CHIRP_1,
+            NATURE_CHANNEL_ANIMALS_4, CHANNEL_IO_PORT_3, 127, // bend pitch
+            NATURE_CHANNEL_ANIMALS_4, CHANNEL_IO_PORT_4, 1,   // number of layers
+            NATURE_CHANNEL_ANIMALS_4, CHANNEL_IO_PORT_5, 16,
+
+            // Channel 6
+            NATURE_CHANNEL_ANIMALS_5, CHANNEL_IO_PORT_2, NATURE_ANIMALS_TAP,
+            NATURE_CHANNEL_ANIMALS_5, CHANNEL_IO_PORT_3, 127, // bend pitch
+            NATURE_CHANNEL_ANIMALS_5, CHANNEL_IO_PORT_4, 3,   // number of layers
+            NATURE_CHANNEL_ANIMALS_5, CHANNEL_IO_PORT_5, 16,
+
+            // Channel 7
+            NATURE_CHANNEL_ANIMALS_6, CHANNEL_IO_PORT_2, NATURE_ANIMALS_CUCCO_CROWS,
+            NATURE_CHANNEL_ANIMALS_6, CHANNEL_IO_PORT_3, 127, // bend pitch
+            NATURE_CHANNEL_ANIMALS_6, CHANNEL_IO_PORT_4, 1,   // number of layers
+            NATURE_CHANNEL_ANIMALS_6, CHANNEL_IO_PORT_5, 16,
+            0xFF
+        }
```

Comment:
I think we should have trailing commas in here. I see we unfortunately disable clang-format, but they could help in some cases either way and it'd be a bit more consistent.
```suggestion
            0xFF,
        },
```

Reply:
The reason clang-format was disabled was because this data is stored just in a raw array of u8, but the data is clearly organized into blocks of 3. It became much less readable when those blocks of three (the channel, port, value) were formatted onto the same line. Did you have any suggestion for that?

![image](https://user-images.githubusercontent.com/47598039/144936556-ae8a6df6-7f75-463e-a406-fe4c78a985a3.png)

Reply:
I think you can have `//` after a group of entries to force a break (every 3 entries here for example)

Reply:
i know we try to avoid clang disables as much as possible, but it seems worse to me to just put a bunch of empty comments when disabling is only two lines
think we should potentially reevaluate our strictness of the format disable at least for cases like this

Reply:
Has the previous policy been to only remove clang formatting if required to match then?

Reply:
There's also meaningful info I could put in comments on each line which would get it to format nicely, at the risk of it being a little cluttered with repetitive comments: For example:
```
            // Channel 1
            NATURE_CHANNEL_ANIMALS_0, CHANNEL_IO_PORT_2, NATURE_ANIMALS_CROWS_CAWS, // sound type
            NATURE_CHANNEL_ANIMALS_0, CHANNEL_IO_PORT_3, 64, // bend pitch
            NATURE_CHANNEL_ANIMALS_0, CHANNEL_IO_PORT_4, 0,  // number of layers
            NATURE_CHANNEL_ANIMALS_0, CHANNEL_IO_PORT_5, 32, // reverb?
```

Reply:
The 2 extra clang-format comments don't really matter at all, but the problem with disabling clang-format is it makes it a pain to work with in the future (both for maintaing the repo and modding), since you can't use the formatter anymore. We're essentially comitting ourselves and everyone else who will work on the code or data to manually format it. The trade off can be worth it in some cases tho, but it usually isn't imo (speaking generally, it might be worth it here for instance).

Reply:
Actually, I think there could be an opportunity to use macros here to really clean things up. Something like
`#define NATURE_ENTRY_ANIMAL_0_TYPE(type) NATURE_CHANNEL_ANIMALS_0, CHANNEL_IO_PORT_2, type,`
`#define NATURE_ENTRY_ANIMAL_0_PITCH(pitch) NATURE_CHANNEL_ANIMALS_0, CHANNEL_IO_PORT_3, pitch,`
`#define NATURE_ENTRY_ANIMAL_0_NUM_LAYERS(num) NATURE_CHANNEL_ANIMALS_0, CHANNEL_IO_PORT_4, num,`
`#define NATURE_ENTRY_ANIMAL_CHAN_0_REVERB_VOLUME(param) NATURE_CHANNEL_ANIMALS_0, CHANNEL_IO_PORT_4, param,`
`#define NATURE_ENTRY_END 0xFF`

Reply:
I'm also considering `ANIMAL` -> `CRITTER` for all these names, thoughts?

Reply:
Alright, so I ended up reworking the macros for these nature entries. There's a lot of macros, so I'm not sure if it's a bit too much. But the data is much easier to read and edit I think, plus I was able to remove the `clang-format off`

Reply:
Wasn't able to look at this for a while but it seems fine to me yeah. The amount of macros is a bit unfortunate but I guess it can't really be avoided with all the different values being used for each entry